pkg/tcpip/transport/udp/protocol.go

```diff
@@ -125,8 +125,9 @@ func (p *protocol) HandleUnknownDestinationPacket(r *stack.Route, id stack.Trans
 		// For example, a raw or packet socket may use what UDP
 		// considers an unreachable destination. Thus we deep copy pkt
 		// to prevent multiple ownership and SR errors.
-		newNetHeader := append(buffer.View(nil), pkt.NetworkHeader...)
-		payload := newNetHeader.ToVectorisedView()
+		newHeader := append(buffer.View(nil), pkt.NetworkHeader...)
+		newHeader = append(newHeader, pkt.TransportHeader...)
+		payload := newHeader.ToVectorisedView()
```

Comment:
do we need a deep copy of the transport header, too?

consider using AppendView here as you do elsewhere.

Reply:
Yes, I'm suggesting that you don't need to do that.

Reply:
The explanation in the comment also applies to transport headers. A view referenced by both `payload` and a raw socket can cause an S/R error, same as network headers.